### ISSUE TITLE
Move Tessellation, Transmission and Translucency from templates to nodes

### ISSUE THREAD

**Issue by diogovtx** [AmplifyCreations/AmplifyShaderEditor-Feedback#137]
Move all tessellation, transmission and translucency code to a node.

- Fix property exposure
- Move code to nodes, when possible
- BiRP, URP and HDRP

**Comment by BOXOPHOBIC** [comment]
Small note, if possible, the nodes should work if used inside functions as well.

**Comment by Dawie3565** [comment]
Note review tessellation code was conducted in ase version. 1.9.1.6
i suggest we move this to node starting from 15x or 16x higher. 

it would be possible to start this in 12x however may break some logic that would need users to add the new node and connect it ... as 15x is still new it will be a good start point for the transition.

**Comment by Creepypie** [comment]
Yeah I'd love seeing this as a node. I'm trying to make my shaders as slim as possible so it'd be great if they could include only stuff that's needed, and in many cases these features aren't used.

**Comment by Dawie3565** [comment]
@diogovtx 
it may be good to land this in 16x just before it comes out of unity beta ?
not certain if we have time in workflow for it before 16x and may get bumped to 17x TBD 

@Creepypie 
the concern is backward compatibility so if we break that its best to have it break to the new method in a fresh api version 
people porting up would need to connect to the new node yes but its an improvement so i think its ok 

we also have a number of other actions to leverage some of the new features in tessellation made by unity into the new nodes

**Comment by Dawie3565** [comment]
Moved this action into its own millstone for workflow.

**Comment by Dawie3565** [comment]
this git item has 3 actions. 

 **Tessellation**
 for this topic we should remove the hook method (SRP) from master blocks and push all the code into tessellation node similar to how it is in BIRP. the code for each pass can be maintained in one location in the node rather than in every template.  
 
 Note: tessellation should also undergo a revamp to aligned with Unity tessellation syntax to leverage this workflow.
 see git item:  https://github.com/AmplifyCreations/AmplifyShaderEditor-Feedback/issues/19 
 check for latest changes, use Unity code for this will help support future improvements in other areas as tessellation chunks can be used in other operations such as height blending
 _TessellationFactor
_TessellationFactorMinDistance
_TessellationFactorMaxDistance
_TessellationFactorTriangleSize
_TessellationShapeFactor
_TessellationBackFaceCullEpsilon

#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Tessellation.hlsl"
#pragma shader_feature_local _ _TESSELLATION_DISPLACEMENT _PIXEL_DISPLACEMENT
#pragma shader_feature_local_domain _TESSELLATION_PHONG
#define TESSELLATION_ON
#define HAVE_TESSELLATION_MODIFICATION

// No tessellation for Meta pass
#undef TESSELLATION_ON

Note:::::: within HeightMap we also use tessellation with displacement
_HeightOffset("Height Offset", Float) = 0


**Transmission & Translucency** can be 2 different outputs but for optimization within the same function. 
this should be very easy to do, and we keep inputs for Transmission & Translucency however only remove the ASE code / hooks within templates. the code for this can be simply put into a shader function so users can modify.  This work is mostly completed and just waiting action to remove the hooks and block code from template. See David for prototype when ready.
